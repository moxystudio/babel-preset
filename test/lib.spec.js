'use strict';

const serializerPath = require('jest-serializer-path');
const preset = require('../lib');

expect.addSnapshotSerializer(serializerPath);

it('should have the right config with the default options', () => {
    expect(preset()).toMatchSnapshot();
});

describe('modules', () => {
    it('should set modules to false if options.modules is disabled', () => {
        expect(preset(null, { modules: false })).toMatchSnapshot();
    });

    it('should set modules to false by default if BABEL_ENV is es', () => {
        process.env.BABEL_ENV = 'es';

        try {
            expect(preset(null)).toMatchSnapshot();
        } finally {
            delete process.env.BABEL_ENV;
        }
    });
});

describe('react', () => {
    it('should enable react plugins if options.react is enabled and enable development goodies', () => {
        expect(preset(null, { react: true })).toMatchSnapshot();
    });

    it('should enable react optimizations if options.env is production', () => {
        expect(preset(null, { env: 'production', react: true })).toMatchSnapshot();
    });
});

describe('lodash', () => {
    it('should disable the lodash plugin if options.lodash is disabled', () => {
        expect(preset(null, { lodash: false })).toMatchSnapshot();
    });

    it('should pass any options to the lodash plugin', () => {
        expect(preset(null, { lodash: { id: 'recompose' } })).toMatchSnapshot();
    });
});

describe('namedDefaultExport & babel-plugin-add-module-exports', () => {
    it('should enable add-module-exports by default', () => {
        expect(preset(null)).toMatchSnapshot();
    });

    it('should enable add-module-exports if options.modules is commonjs', () => {
        expect(preset(null)).toMatchSnapshot();
        expect(preset(null, { modules: 'commonjs' })).toMatchSnapshot();
    });

    it('should not enable add-module-exports if options.modules is not commonjs', () => {
        expect(preset(null, { modules: false })).toMatchSnapshot();
    });

    it('should respect options.namedDefaultExport', () => {
        expect(preset(null, { namedDefaultExport: true })).toMatchSnapshot();
        expect(preset(null, { modules: 'commonjs', namedDefaultExport: true })).toMatchSnapshot();
        expect(preset(null, { modules: 'cjs', namedDefaultExport: true })).toMatchSnapshot();
        expect(preset(null, { namedDefaultExport: false })).toMatchSnapshot();
        expect(preset(null, { modules: 'commonjs', namedDefaultExport: false })).toMatchSnapshot();
        expect(preset(null, { modules: 'cjs', namedDefaultExport: false })).toMatchSnapshot();
    });

    it('should throw when enabling options.namedDefaultExport and options.modules is not commonjs', () => {
        expect(() => preset(null, { modules: false, namedDefaultExport: true }))
        .toThrow('The `namedDefaultExport` option can only be enabled when `modules` is commonjs');
    });
});

describe('dynamicImport', () => {
    it('should respect options.modules and options.dynamicImport', () => {
        expect(preset(null, { modules: 'commonjs', dynamicImport: true })).toMatchSnapshot();
        expect(preset(null, { modules: 'cjs', dynamicImport: true })).toMatchSnapshot();
        expect(preset(null, { modules: false, dynamicImport: true })).toMatchSnapshot();
        expect(preset(null, { modules: 'foo', dynamicImport: true })).toMatchSnapshot();

        expect(preset(null, { modules: 'commonjs', dynamicImport: false })).toMatchSnapshot();
        expect(preset(null, { modules: 'cjs', dynamicImport: false })).toMatchSnapshot();
        expect(preset(null, { modules: false, dynamicImport: false })).toMatchSnapshot();
        expect(preset(null, { modules: 'foo', dynamicImport: true })).toMatchSnapshot();
    });
});
