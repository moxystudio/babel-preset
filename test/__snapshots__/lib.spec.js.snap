// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamicImport should respect options.modules and options.dynamicImport 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`dynamicImport should respect options.modules and options.dynamicImport 2`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`dynamicImport should respect options.modules and options.dynamicImport 3`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`dynamicImport should respect options.modules and options.dynamicImport 4`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`dynamicImport should respect options.modules and options.dynamicImport 5`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`dynamicImport should respect options.modules and options.dynamicImport 6`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`dynamicImport should respect options.modules and options.dynamicImport 7`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`dynamicImport should respect options.modules and options.dynamicImport 8`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`lodash should disable the lodash plugin if options.lodash is disabled 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`lodash should pass any options to the lodash plugin 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
          "recompose",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`modules should set modules to false by default if BABEL_ENV is es 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`modules should set modules to false if options.modules is disabled 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should enable add-module-exports by default 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should enable add-module-exports if options.modules is commonjs 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should enable add-module-exports if options.modules is commonjs 2`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should not enable add-module-exports if options.modules is not commonjs 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 2`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 3`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 4`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 5`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 6`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [],
}
`;

exports[`react should enable react optimizations if options.env is production 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-display-name/lib/index.js",
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`react should enable react plugins if options.react is enabled and enable development goodies 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-display-name/lib/index.js",
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;

exports[`should have the right config with the default options 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-modules-commonjs",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [],
}
`;
